utils/measure/measure/measure.py

```diff
@@ -272,6 +274,22 @@ def ask_questions(self, questions: list[Question]) -> dict[str, Any]:
         answers = inquirer.prompt(questions_to_ask, answers=predefined_answers, render=self.console_render)
         answers.update(predefined_answers)
 
+        # if either no dummy load is used or not with hass power meters, skip this
+        if not (answers.get(QUESTION_DUMMY_LOAD, False) and isinstance(self.power_meter, HassPowerMeter)):
+            pass
+        elif self.power_meter.autodetect_voltage_entity(answers.get(QUESTION_POWERMETER_ENTITY_ID, None)):
+            # Autodetection succeeded
+            pass
+        else:
+            # Ask voltage question if autodetection failed
+            voltage_question = self.power_meter.get_voltage_question()
+            if voltage_question:
+                voltage_answer = self.ask_questions(voltage_question)
+                answers.update(voltage_answer)
+            else:
+                _LOGGER.error("Voltage Meter question result was empty")
+                exit(1)
```

Comment:
btw I just notice there is several code added now specifically for powermeters, and even more specifically for the hass powermeter only.

There is specific extension point for this designed. See line 248.
Powermeter protocol has `get_questions` method where you can extend the questions with more questions in `hass.py`.